src/objective-c/GRPCClient/GRPCCall.m

```diff
@@ -354,8 +357,29 @@ - (void)startWithWriteable:(id<GRXWriteable>)writeable {
   _retainSelf = self;
 
   _responseWriteable = [[GRXConcurrentWriteable alloc] initWithWriteable:writeable];
+
+  _wrappedCall = [[GRPCWrappedCall alloc] initWithHost:_host path:_path];
+  NSAssert(_wrappedCall, @"Error allocating RPC objects. Low memory?");
+
   [self sendHeaders:_requestHeaders];
   [self invokeCall];
+  // TODO(jcanizales): Extract this logic somewhere common.
+  NSString *host = [NSURL URLWithString:[@"https://" stringByAppendingString:_host]].host;
+  if (!host) {
+    // TODO(jcanizales): Check this on init.
+    [NSException raise:NSInvalidArgumentException format:@"host of %@ is nil", _host];
+  }
+  __weak typeof(self) weakSelf = self;
+  _connectivityMonitor = [GRPCConnectivityMonitor monitorWithHost:host];
```

Comment:
I've not used `SCNetworkReachabilityCreateWithAddress` much before, but the uses of it in Chrome/Firefox seem to use `0.0.0.0` as their host target, does it make sense to use this too, of the host for this specific call?

https://code.google.com/p/chromium/codesearch#chromium/src/net/base/network_change_notifier_mac.cc&q=SCNetworkReachabilityCreateWithAddress&sq=package:chromium&type=cs&l=133